pkg/helm/chartproxy/proxy_test.go

```diff
@@ -1,31 +1,47 @@
 package chartproxy
 
 import (
-	"golang.org/x/net/context"
-	"helm.sh/helm/v3/pkg/repo"
 	"io/ioutil"
-	v1 "k8s.io/apimachinery/pkg/apis/meta/v1"
-	"k8s.io/apimachinery/pkg/apis/meta/v1/unstructured"
 	"reflect"
 	"testing"
 
+	"golang.org/x/net/context"
+	"helm.sh/helm/v3/pkg/repo"
+	v1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/apis/meta/v1/unstructured"
 	"k8s.io/client-go/rest"
 
 	"github.com/openshift/console/pkg/helm/actions/fake"
 )
 
 func TestProxy_IndexFile(t *testing.T) {
 	tests := []struct {
-		name       string
-		indexFiles []string
-		mergedFile string
-		helmCRS    []*unstructured.Unstructured
+		name           string
+		indexFiles     []string
+		mergedFile     string
+		kubeVersion    string
+		helmCRS        []*unstructured.Unstructured
+		onlyCompatible bool
 	}{
 		{
 			name:       "returned index file for configured helm repo",
 			indexFiles: []string{"testdata/sampleRepoIndex.yaml"},
 			mergedFile: "testdata/mergedSampleRepoIndex2.yaml",
 		},
+		{
+			name:           "returned index file for configured helm repo contains only charts compatible for given cluster",
+			indexFiles:     []string{"testdata/sampleRepoIndex3.yaml"},
+			mergedFile:     "testdata/sampleRepoIndex3Filtered.yaml",
+			kubeVersion:    "v1.16.0",
+			onlyCompatible: true,
+		},
+		{
+			name:           "returned empty index file for configured helm repo contains only charts compatible for given cluster",
```

Comment:
let's name it as `return empty index file if not charts are compatible with given cluster`

Reply:
Done